gprofiler/utils/node.py

```diff
@@ -0,0 +1,243 @@
+import json
+import os
+import shutil
+import signal
+import stat
+from functools import lru_cache
+from threading import Event
+from typing import Dict, List, cast
+
+import psutil
+import requests
+from granulate_utils.linux.ns import get_proc_root_path, get_process_nspid, resolve_proc_root_links, run_in_ns
+from granulate_utils.linux.process import is_musl, is_process_running
+from retry import retry
+from websocket import create_connection
+from websocket._core import WebSocket
+
+from gprofiler.log import get_logger_adapter
+from gprofiler.metadata.versions import get_exe_version
+from gprofiler.utils import add_permission_dir, pgrep_exe, resource_path
+
+logger = get_logger_adapter(__name__)
+
+
+class NodeDebuggerUrlNotFound(Exception):
+    pass
+
+
+class NodeDebuggerUnexpectedResponse(Exception):
+    pass
+
+
+def _get_node_major_version(process: psutil.Process) -> str:
+    node_version = get_exe_version(process, Event(), 3)
+    # i. e. v16.3.2 -> 16
+    return node_version[1:].split(".")[0]
+
+
+def _get_dso_git_rev() -> str:
+    libc_dso_version_file = resource_path(os.path.join("node", "module", "glibc", "version"))
+    musl_dso_version_file = resource_path(os.path.join("node", "module", "musl", "version"))
+    with open(libc_dso_version_file) as f:
+        libc_dso_ver = f.readline()
+    with open(musl_dso_version_file) as f:
+        musl_dso_ver = f.readline()
+    # with no build errors this should always be the same
+    assert libc_dso_ver == musl_dso_ver
+    return libc_dso_ver
+
+
+@lru_cache()
+def _get_dest_inside_container(musl: bool, node_version: str) -> str:
+    libc = "musl" if musl else "glibc"
+    return os.path.join("/", "tmp", "node_module", _get_dso_git_rev(), libc, node_version)
```

Comment:
Please import `GPROFILER_DIRECTORY_NAME` and use it as a path component here (to avoid placing files outside of gprofiler directories)